x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/InferenceProcessor.java

```diff
@@ -342,6 +373,7 @@ public void accept(ClusterState state) {
         }
 
         @Override
+        @SuppressWarnings("unchecked")
```

Comment:
How come we need the unchecked suppression here?

Reply:
The code parsing the raw maps requires it. Refactoring that out to new method as you suggested means it can now be removed